### ISSUE TITLE
Some problems in reproducing

### ISSUE THREAD

**Issue by flashingtt** [nasimjamshidi/LAR-IQA#11]
@nasimjamshidi Hi, thank you for your excellent work!
I have some problems trying to reproduce the results in the paper. I wonder if there is something wrong with my implementation.
Here are some details of my scrips:
```
python scripts/main.py \
    --csv_files dataset/full_training.csv \
    --root_dirs datasets/IQA/UHD-IQA-database/training \
    --val_csv_file dataset/full_validation.csv \
    --val_root_dir datasets/IQA/UHD-IQA-database/validation \
    --output_path output.csv \
    --num_epochs 100 \
    --batch_size 32 \
    --use_kan \
    --color_space RGB \
    --use_sweep
```
I split the `uhd-iqa-metadata.csv` file in the UHD-IQA dataset into `uhd-iqa-training-metadata.csv` and `uhd-iqa-validation-metadata.csv` for training and validation respectively. `full_training.csv` and `full_validation.csv` have 4270 and 905 data respectively, replacing the original `training.csv` and `validation.csv` in the repo.
Here is the settings of the sweep_config, I change the num_epochs to 100:
```
sweep_configuration = {
            'method': 'random',
            'metric': {'name': 'val_loss', 'goal': 'minimize'},
            'parameters': {
                'learning_rate': {'values': [5e-5]},
                'weight_decay': {'values': [1e-4]},
                'warmup_epochs': {'values': [5]},
                'num_epochs': {'values': [100]},
                'l_num_epochs': {'values': [10]},
                'optimizer': {'values': ['adamw']},
                'NR_msel_weight': {'values': [1.0]},
                'NR_crl_weight': {'values': [1.0]}
            }
        }

```
Because of the code
```
train.py:55:    max_iter = int((config.num_epochs + config.l_num_epochs) * len(train_loader))
```
the entire training time is too long, and the loss on the validation set converges to about 0.6 at 6k steps.
Could you please give some tips?

**Comment by nasimjamshidi** [comment]
@flashingtt : Hi, thank you for your detailed feedback and for trying to reproduce the results!

I’ve found that in the last update, I mistakenly forgot to load the pretrained models for the KAN version. As a result, the model was training from scratch, which explains why it didn’t learn as expected and why the validation loss plateaued early. While this oversight caused trouble for your experiments, it also highlights how important it is to pretrain each branch (authentic and synthetic) separately — this pretraining significantly boosts performance and convergence.

I’ve fixed this issue in the train.py file — specifically at line 133 — so you can pull the latest changes and try running the training again using the same script and config.

Regarding the long training time:
This is primarily due to I/O overhead (due to 4k images) and the large patch size (1280×1280), which is computationally heavy. Additionally, initializing and training the KAN model introduces some extra latency. You might consider using smaller patches or optimizing data loading (e.g. with caching or faster storage) if training time becomes a bottleneck.

Please let me know if you face any more issues — and thank you again for pointing this out!

**Comment by flashingtt** [comment]
@nasimjamshidi Hi, the training situation has improved after loading the pre-trained model. Thank you for your patience and detailed reply!
